tree/tree/src/TBranchElement.cxx

```diff
@@ -2416,7 +2475,17 @@ void TBranchElement::InitInfo()
                localInfo = FindOnfileInfo(fClonesClass, fBranches);
             }
 
-            TString prefix(GetName());
+            TString prefix(GetFullName());
+            if (fType == 2 && fID >= 0) {
+               auto start = prefix.Length();
+               if (prefix[prefix.Length()-1] == '.')
```

Comment:
```suggestion
               if (prefix[start-1] == '.')
```
(Possibly more obvious given the `--start` right after)

Reply:
ok.